tests/unit/services/cloudformation/test_provider_utils.py

```diff
@@ -0,0 +1,12 @@
+import localstack.services.cloudformation.provider_utils as utils
+
+
+class TestDictUtils:
+    def test_convert_values_to_numbers(self):
+        original = {"Parameter": "1", "SecondParameter": ["2", "2"], "ThirdParameter": "3"}
+        transformed = utils.convert_values_to_numbers(original, ["ThirdParameter"])
+
+        assert original["Parameter"] == "1"
+        assert transformed["Parameter"] == 1
+        assert transformed["SecondParameter"][0] == 2
+        assert transformed["ThirdParameter"] == "3"
```

Comment:
nit: could you compare the transformed with a manually created version of the original, then it's clearer for readers to see the difference

```suggestion
        assert transformed == {"Parameter": 1, "SecondParameter": [2, 2], "ThirdParameter": "3"}
```